schema.sql

```diff
@@ -10,75 +10,84 @@ DROP TABLE IF EXISTS country CASCADE;
 
 
 CREATE TABLE country (
-    country_id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    country_id SMALLINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
     country_name VARCHAR UNIQUE NOT NULL
 );
 
 
 CREATE TABLE customer (
-    customer_id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    customer_id SMALLINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
     customer_name VARCHAR NOT NULL,
-    country_id BIGINT,
+    country_id SMALLINT,
     FOREIGN KEY (country_id) REFERENCES country(country_id)
 );
 
 
 CREATE TABLE artist (
-    artist_id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    artist_id SMALLINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
     artist_name VARCHAR NOT NULL,
-    country_id BIGINT,
+    country_id SMALLINT,
     FOREIGN KEY (country_id) REFERENCES country(country_id)
 );
 
 
 CREATE TABLE genre (
-    genre_id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    genre_id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
     genre_name VARCHAR NOT NULL
 );
 
 
 CREATE TABLE type (
-    type_id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    type_id SMALLINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
     type_name VARCHAR NOT NULL
 );
 
 
 CREATE TABLE release (
-    release_id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    release_id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
     release_name VARCHAR NOT NULL,
     release_date DATE NOT NULL,
-    genre_id BIGINT,
-    artist_id BIGINT,
-    type_id BIGINT,
-    FOREIGN KEY (genre_id) REFERENCES genre(genre_id),
+    artist_id SMALLINT,
+    type_id SMALLINT,
     FOREIGN KEY (artist_id) REFERENCES artist(artist_id),
     FOREIGN KEY (type_id) REFERENCES type(type_id)
 );
 
+CREATE TABLE release_genre (
+    release_id INT,
+    genre_id INT,
+    PRIMARY KEY (release_id, genre_id),
+    FOREIGN KEY (release_id) REFERENCES release(release_id) ON DELETE CASCADE,
+    FOREIGN KEY (genre_id) REFERENCES genre(genre_id) ON DELETE CASCADE
+);
 
 CREATE TABLE sale (
-    sale_id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    sale_id SMALLINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
     sale_price FLOAT NOT NULL,
     sale_date DATE NOT NULL,
-    customer_id BIGINT,
-    release_id BIGINT,
+    customer_id SMALLINT,
```

Comment:
wondering whether this should be int too? just because it's likely we might have more than 32000 customers